mlflow/tracking/fluent.py

```diff
@@ -3049,3 +3050,74 @@ def setup_autologging(module):
             register_post_import_hook(setup_autologging, "pyspark", overwrite=True)
         if "pyspark.ml" in target_library_and_module:
             register_post_import_hook(setup_autologging, "pyspark.ml", overwrite=True)
+
+
+_ACTIVE_MODEL_ID = ContextVar("active_model_id", default=None)
+
+
+class ActiveModel(LoggedModel):
+    """
+    Wrapper around :py:class:`mlflow.entities.LoggedModel` to enable using Python ``with`` syntax.
+    """
+
+    def __init__(self, logged_model: LoggedModel):
+        super().__init__(**logged_model.to_dictionary())
+        self.last_active_model_id = _ACTIVE_MODEL_ID.get()
+        _ACTIVE_MODEL_ID.set(self.model_id)
+        _logger.info(f"Active model set to model with ID: {self.model_id}")
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        _ACTIVE_MODEL_ID.set(self.last_active_model_id)
+
+
+def set_active_model(*, name: Optional[str] = None, model_id: Optional[str] = None) -> ActiveModel:
+    if name is None and model_id is None:
+        raise MlflowException.invalid_parameter_value(
+            message="Either name or model_id must be provided",
+        )
+
+    if model_id is not None:
+        try:
+            logged_model = mlflow.get_logged_model(model_id)
+        except Exception as e:
+            raise MlflowException(f"Failed to find LoggedModel with model_id {model_id}") from e
```

Comment:
Do we need this try-catch?

Reply:
I want to unify different error messages from different backends, and print a more informative message for this API, and it doesn't seem like we're hiding any errors from users?

Reply:
Should we update `get_logged_model` then?